### ISSUE TITLE
Map & list variable values loaded from env variables trigger "Incorrect variable type" warning with module inspection

### ISSUE THREAD

**Issue by darrengrove-RL** [terraform-linters/tflint#1281]
If you have a variable defined as a map or list:

```hcl
variable "map_variable" {
  type        = map(string)
  description = "Map of string to strings"
  default     = {}
}
```

And you have it loaded from environment variables such as:
`export TF_VAR_map_variable='{ "test": "post" }'`

And passed into a module as:
```hcl
module "test_variable" {
  source             = "./test_variable"
  map_variable       = var.map_variable
}
```

It results in an error with `tflint --module`
```hcl
Error: Failed to eval an expression in main.tf:35; Incorrect variable type: The resolved value of variable "map_variable" is not appropriate: map of string required.
```

If you unset the environment variable you no longer receive the error:
`unset TF_VAR_map_variable`

Terraform itself applies with the environment variable without issue. I haven't been able to set the variable with any other type of formatting without triggering the error.

## Version
0.33.1

From debug:
`14:43:57 terraform.go:158: [INFO] TF_VAR_* environment variable found: key=TF_VAR_map_variable`

**Comment by wata727** [comment]
Thank you for opening this issue. Currently, TFLint parses `TF_VAR_*` env variables as string literals. So this error has occurred.
This is inconsistent with the recent Terraform behavior and we are considering fixing it.

**Comment by wata727** [comment]
v0.34.1 has been released.
https://github.com/terraform-linters/tflint/releases/tag/v0.34.1
